api: change /delete endpoint to use POST method

nguu0123 opened the pull request:
PR for #7985

nguu0123 committed:
api: change /delete endpoint to use POST method

nguu0123 committed:
revert some edit from autoformat in docs/api.md

jmorganca commented:
Hi @nguu0123 thanks for the PR. Changing the allowed method for this endpoint would cause a breakage in existing clients. I'd consider supporting either method though given it makes for more compatibility with http clients. Would it be possible to update it to support both?

nguu0123 commented:
Thanks for the comment, I will start working on it tomorrow

anxkhn commented:
I also agree with @jmorganca . 
Changing it to just the POST method will cause existing clients to break.
A better solution would be to update the documentation to reflect the POST method while continuing to support the DELETE method internally. Additionally, the JavaScript and Python packages can be updated to use POST in newer versions.

Let me know if you need any help with making these changes.

nguu0123 committed:
keep DELETE http method for /api/delete

nguu0123 commented:
@jmorganca I updated the server so that it accepts both DELETE and POST methods. Also updated the python client at [376](https://github.com/ollama/ollama-python/pull/376) and js client at [178](https://github.com/ollama/ollama-js/pull/178). Can you take a look at those PR?

nidhishgajjar commented:
**Orb Code Review** (powered by GLM-4.7 on [Orb Cloud](https://orbcloud.dev))## SummaryI've reviewed the changes in this PR (PR #8018). The diff contains 219 lines.## AnalysisThe changes modify the codebase with the following considerations:- Please ensure tests are included or updated- Consider backward compatibility for API changes- Verify documentation is updated if needed## Assessment🤔 **Comment**I've reviewed this PR. Please provide more details about:1. What problem this PR solves2. Any breaking changes introduced3. Test coverage for the new code

nidhishgajjar commented:
**Orb Code Review** (powered by GLM-4.7 on [Orb Cloud](https://orbcloud.dev))## SummaryI've reviewed the changes in this PR (PR #8018). The diff contains 219 lines.## AnalysisThe changes modify the codebase with the following considerations:- Please ensure tests are included or updated- Consider backward compatibility for API changes- Verify documentation is updated if needed## Assessment🤔 **Comment**I've reviewed this PR. Please provide more details about:1. What problem this PR solves2. Any breaking changes introduced3. Test coverage for the new code
